source/base/parameter_handler.cc

```diff
@@ -641,37 +641,28 @@ namespace
         if (p.second.empty())
           {
             // set the found parameter in the destination argument
-            if (skip_undefined)
+            try
               {
-                try
-                  {
-                    prm.set(demangle(p.first), p.second.data());
-                  }
-                catch (const ParameterHandler::ExcEntryUndeclared &)
-                  {
-                    // ignore undeclared entry assert
-                  }
+                prm.set(demangle(p.first), p.second.data());
+              }
+            catch (const ParameterHandler::ExcEntryUndeclared &e)
+              {
+                // ignore undeclared entry assert
+                AssertThrow(skip_undefined, e);
```

Comment:
Ah, clever :-)

Reply:
The described issue should be fixed with the last commit.